src/stores/events/EventsDataStore.ts

```diff
@@ -165,7 +170,10 @@ export default class EventsDataStore {
 			if (event.parentId === null) {
 				rootEvents.push(event);
 			} else {
-				eventsByParentId[event.parentId] = (eventsByParentId[event.parentId] || []).concat(event);
+				const cachedParent = this.eventsCache.get(event.parentId);
+				// We need this to change the parent to an unknown root if needed
+				const parentId = cachedParent?.eventId ?? event.parentId;
```

Comment:
if `cachedParent.eventId` and `event.parentId` is the same string, why do we need to even get parent from cache and check it?

Reply:
`CachedParent.eventId` may be `'unknown-root'` because cache contains unknown group if event wasn't found